src/routes/v0/tag/global.rs

```diff
@@ -75,53 +78,74 @@ pub async fn tag_taggers_handler(
     Query(query): Query<TagTaggersQuery>,
 ) -> Result<Json<Vec<String>>> {
     info!(
-        "GET {TAG_TAGGERS_ROUTE} label:{}, reach:{:?}",
-        label, query.reach
+        "GET {TAG_TAGGERS_ROUTE} label:{}, query: {:?}",
+        label, query
     );
 
-    let reach = query.reach;
-
-    match TagGlobal::get_tag_taggers(label.clone(), reach).await {
+    match TagGlobal::get_tag_taggers(
+        label.clone(),
+        query.user_id,
+        query.reach,
+        query.pagination.skip,
+        query.pagination.limit,
+    )
+    .await
+    {
         Ok(Some(post)) => Ok(Json(post)),
         Ok(None) => Err(Error::TagsNotFound { reach: label }),
         Err(source) => Err(Error::InternalServerError { source }),
     }
 }
 
-#[derive(Deserialize)]
+#[derive(Deserialize, Debug)]
 pub struct TagsByReachQuery {
+    skip: Option<usize>,
+    limit: Option<usize>,
+    max_taggers: Option<usize>,
+}
+
+#[derive(Deserialize)]
+pub struct TagsByReachPath {
     user_id: String,
     reach: Option<TagStreamReach>,
 }
 
 #[utoipa::path(
     get,
-    path = TAG_REACH_ROUTE,
-    tag = "Global Tags by reach",
+    path = HOT_TAGS_BY_REACH_ROUTE,
+    tag = "Hot Tags by reach",
     params(
         ("user_id" = String, Path, description = "User Pubky ID"),
-        ("reach" = TagStreamReach, Path, description = "Reach type: Follower | Following | Friends")
+        ("reach" = TagStreamReach, Path, description = "Reach type: Follower | Following | Friends"),
+        ("skip" = Option<usize>, Query, description = "Skip N tags"),
+        ("limit" = Option<usize>, Query, description = "Retrieve N tag"),
+        ("max_taggers" = Option<usize>, Query, description = "Retrieve N user_id for each tag")
     ),
     responses(
         (status = 200, description = "Retrieve tags by reach cluster", body = Vec<HotTag>),
         (status = 404, description = "Hot tags not found"),
         (status = 500, description = "Internal server error")
     )
 )]
-pub async fn tags_by_reach_handler(Path(path): Path<TagsByReachQuery>) -> Result<Json<HotTags>> {
+pub async fn tags_by_reach_handler(
+    Path(path): Path<TagsByReachPath>,
+    Query(query): Query<TagsByReachQuery>,
+) -> Result<Json<HotTags>> {
     info!(
-        "GET {TAG_REACH_ROUTE}: {:?}, {:?}",
-        path.user_id, path.reach
+        "GET {HOT_TAGS_BY_REACH_ROUTE} user_id: {:?}, reach: {:?}, query: {:?}",
+        path.user_id, path.reach, query
     );
 
     let reach = path.reach.unwrap_or(TagStreamReach::Following);
     let user_id = path.user_id;
 
-    match HotTags::get_stream_tags_by_reach(user_id, reach.clone()).await {
+    let skip = query.skip.unwrap_or(0);
+    let limit = query.limit.unwrap_or(20);
+    let max_taggers = query.max_taggers.unwrap_or(20);
```

Comment:
add default maximun value

Reply:
it's there?

Reply:
no. It is possible if the user sets `max_taggers` and `limit` params with 100, there is not limit if the user provides the value
```rust
 let limit = query.pagination.limit.unwrap_or(20).min(50);
```